src/CosmosDB/CosmosDB/SQL/NewAzCosmosDbClientEncryptionKey.cs

```diff
@@ -0,0 +1,180 @@
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.EncryptionAlgorithmName)]
+        [ValidateNotNullOrEmpty]
+        public string EncryptionAlgorithmName { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = Constants.KeyWrapMetaData)]
+        [ValidateNotNullOrEmpty]
+        public PSSqlKeyWrapMetadata KeyWrapMetadata { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipeline = true, HelpMessage = Constants.WrappedDataEncryptionKey)]
+        [ValidateNotNullOrEmpty]
+        public byte[] WrappedDataEncryptionKey { get; set; }        
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.IsAzureKeyVaultKeyStoreProvider)]
+        [ValidateNotNullOrEmpty]
+        public bool IsAzureKeyVaultKeyStoreProvider { get; set; } = true;
+
+        [Parameter(Mandatory = false, ValueFromPipeline = true, HelpMessage = Constants.IsAzureKeyVaultKeyStoreProvider)]
+        [ValidateNotNullOrEmpty]
+        public EncryptionKeyStoreProvider EncryptionKeyStoreProvider { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParentObjectParameterSet, HelpMessage = Constants.SqlDatabaseObjectHelpMessage)]
+        [ValidateNotNull]
+        public PSSqlDatabaseGetResults ParentObject { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(ParentObjectParameterSet, StringComparison.Ordinal))
+            {
+                ResourceIdentifier resourceIdentifier = new ResourceIdentifier(ParentObject.Id);
+                ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                DatabaseName = resourceIdentifier.ResourceName;
+                AccountName = ResourceIdentifierExtensions.GetDatabaseAccountName(resourceIdentifier);
+            }
+
+            KeyWrapMetadata encryptionKeyWrapMetadata;
+            if (KeyWrapMetadata != null)
+            {
+                encryptionKeyWrapMetadata = PSSqlKeyWrapMetadata.ToSDKModel(KeyWrapMetadata);
+            }
+            else
+            {
+                throw new ArgumentException("KeyWrapMetadata cannot be null");
+            }
+
+            ClientEncryptionKeyGetResults readClientEncryptionKeyGetResults = null;
+            try
+            {
+                readClientEncryptionKeyGetResults = CosmosDBManagementClient.SqlResources.GetClientEncryptionKey(ResourceGroupName, AccountName, DatabaseName, ClientEncryptionKeyName);
+            }
+            catch (CloudException e)
+            {
+                if (e.Response.StatusCode != System.Net.HttpStatusCode.NotFound)
+                {
+                    throw;
+                }
+            }
+
+            if (readClientEncryptionKeyGetResults != null)
+            {
+                throw new ConflictingResourceException(message: string.Format(ExceptionMessage.Conflict, ClientEncryptionKeyName));
+            }
```

Comment:
Why do we need to do this explicitly rather than the create itself handling it (given due to races that can anyway happen)?

Reply:
the Mgmt SDK exposes CreateUpdateClientEncryptionKeyWithHttpMessagesAsync hence need to check for this. All other Cmdlets (creating container etc) have CreateUpdate.

Reply:
need to revisit this.

Reply:
as discussed. Moved the check for existing resource  to a later stage(post key creation and wrapping) and just before the PUT request.